common/flatpak-run.c

```diff
@@ -1597,15 +1597,26 @@ add_tzdata_args (FlatpakBwrap *bwrap,
   g_autofree char *timezone_content = g_strdup_printf ("%s\n", raw_timezone);
   g_autofree char *localtime_content = g_strconcat ("../usr/share/zoneinfo/", raw_timezone, NULL);
   g_autoptr(GFile) runtime_zoneinfo = NULL;
+  const gchar *tzdir;
 
   if (runtime_files)
     runtime_zoneinfo = g_file_resolve_relative_path (runtime_files, "share/zoneinfo");
 
   /* Check for runtime /usr/share/zoneinfo */
   if (runtime_zoneinfo != NULL && g_file_query_exists (runtime_zoneinfo, NULL))
     {
+      /* Check for host TZDIR */
+      tzdir = getenv ("TZDIR");
+      if (tzdir != NULL && g_file_test (tzdir, G_FILE_TEST_IS_DIR))
+        {
+          /* Here we assume the host timezone file exist in the host data */
+          flatpak_bwrap_add_args (bwrap,
+                                  "--ro-bind", tzdir, "/usr/share/zoneinfo",
```

Comment:
I cannot really test this at the moment. One possible complication is that on NixOS, the path is a symlink to symlink and I am not sure how bubblewrap will treat this.

```ShellSession
$ echo $TZDIR
/etc/zoneinfo
$ readlink $TZDIR
/etc/static/zoneinfo
$ readlink $(readlink $TZDIR)
/nix/store/y6hmqbmbwq0rmx1fzix5c5jszla2pzmp-tzdata-2024a/share/zoneinfo
```

Reply:
I don't know either, and haven't tested, but I think it will work normally

Reply:
Looks like the symlinks passed to `--ro-bind` are dereferenced:

```ShellSession
$ mkdir -p test/{etc/static,data/zoneinfo}
$ touch test/data/zoneinfo/UTC
$ ln -s $PWD/test/etc/static/zoneinfo test/etc
$ ln -s $PWD/test/data/zoneinfo test/etc/static
$ ln -s ~/Projects/gtk/COPYING test/etc/static/zoneinfo/

$ exa -T test
test
├── data
│  └── zoneinfo
│     ├── COPYING -> /home/jtojnar/Projects/gtk/COPYING
│     └── UTC
└── etc
   ├── static
   │  └── zoneinfo -> /home/jtojnar/test/data/zoneinfo
   └── zoneinfo -> /home/jtojnar/test/etc/static/zoneinfo

$ bwrap \
   --ro-bind-try /nix/store /nix/store \
   --ro-bind-try /usr /usr \
   --symlink usr/lib64 /lib64 \
   --ro-bind $PWD/test/etc/zoneinfo /foo/zoneinfo \
   --proc /proc \
   --dev /dev \
   --unshare-pid \
   --new-session \
   bash -c 'ls -laR /foo/zoneinfo'

/foo/zoneinfo:
total 4
drwxr-xr-x 2 1000 100 4096 Jun 20 13:47 .
drwx------ 3 1000 100   60 Jun 20 13:48 ..
lrwxrwxrwx 1 1000 100   34 Jun 20 13:47 COPYING -> /home/jtojnar/Projects/gtk/COPYING
-rw-r--r-- 1 1000 100    0 Jun 20 13:42 UTC
```

It does not do anything about the symlinks inside the bind-mount but it looks like `$TZDIR` currently only uses relative symlinks so it should be fine:

<details>
<summary>NixOS `$TZDIR` directory tree</summary>

```
$ exa -T $TZDIR/
/etc/zoneinfo
├── Africa
│  ├── Abidjan
│  ├── Accra
│  ├── Addis_Ababa
│  ├── Algiers
│  ├── Asmara
│  ├── Asmera
│  ├── Bamako
│  ├── Bangui
│  ├── Banjul
│  ├── Bissau
│  ├── Blantyre
│  ├── Brazzaville
│  ├── Bujumbura
│  ├── Cairo
│  ├── Casablanca
│  ├── Ceuta
│  ├── Conakry
│  ├── Dakar
│  ├── Dar_es_Salaam
│  ├── Djibouti
│  ├── Douala
│  ├── El_Aaiun
│  ├── Freetown
│  ├── Gaborone
│  ├── Harare
│  ├── Johannesburg
│  ├── Juba
│  ├── Kampala
│  ├── Khartoum
│  ├── Kigali
│  ├── Kinshasa
│  ├── Lagos
│  ├── Libreville
│  ├── Lome
│  ├── Luanda
│  ├── Lubumbashi
│  ├── Lusaka
│  ├── Malabo
│  ├── Maputo
│  ├── Maseru
│  ├── Mbabane
│  ├── Mogadishu
│  ├── Monrovia
│  ├── Nairobi
│  ├── Ndjamena
│  ├── Niamey
│  ├── Nouakchott
│  ├── Ouagadougou
│  ├── Porto-Novo
│  ├── Sao_Tome
│  ├── Timbuktu
│  ├── Tripoli
│  ├── Tunis
│  └── Windhoek
├── America
│  ├── Adak
│  ├── Anchorage
│  ├── Anguilla
│  ├── Antigua
│  ├── Araguaina
│  ├── Argentina
│  │  ├── Buenos_Aires
│  │  ├── Catamarca
│  │  ├── ComodRivadavia
│  │  ├── Cordoba
│  │  ├── Jujuy
│  │  ├── La_Rioja
│  │  ├── Mendoza
│  │  ├── Rio_Gallegos
│  │  ├── Salta
│  │  ├── San_Juan
│  │  ├── San_Luis
│  │  ├── Tucuman
│  │  └── Ushuaia
│  ├── Aruba
│  ├── Asuncion
│  ├── Atikokan
│  ├── Atka
│  ├── Bahia
│  ├── Bahia_Banderas
│  ├── Barbados
│  ├── Belem
│  ├── Belize
│  ├── Blanc-Sablon
│  ├── Boa_Vista
│  ├── Bogota
│  ├── Boise
│  ├── Buenos_Aires
│  ├── Cambridge_Bay
│  ├── Campo_Grande
│  ├── Cancun
│  ├── Caracas
│  ├── Catamarca
│  ├── Cayenne
│  ├── Cayman
│  ├── Chicago
│  ├── Chihuahua
│  ├── Ciudad_Juarez
│  ├── Coral_Harbour
│  ├── Cordoba
│  ├── Costa_Rica
│  ├── Creston
│  ├── Cuiaba
│  ├── Curacao
│  ├── Danmarkshavn
│  ├── Dawson
│  ├── Dawson_Creek
│  ├── Denver
│  ├── Detroit
│  ├── Dominica
│  ├── Edmonton
│  ├── Eirunepe
│  ├── El_Salvador
│  ├── Ensenada
│  ├── Fort_Nelson
│  ├── Fort_Wayne
│  ├── Fortaleza
│  ├── Glace_Bay
│  ├── Godthab
│  ├── Goose_Bay
│  ├── Grand_Turk
│  ├── Grenada
│  ├── Guadeloupe
│  ├── Guatemala
│  ├── Guayaquil
│  ├── Guyana
│  ├── Halifax
│  ├── Havana
│  ├── Hermosillo
│  ├── Indiana
│  │  ├── Indianapolis
│  │  ├── Knox
│  │  ├── Marengo
│  │  ├── Petersburg
│  │  ├── Tell_City
│  │  ├── Vevay
│  │  ├── Vincennes
│  │  └── Winamac
│  ├── Indianapolis
│  ├── Inuvik
│  ├── Iqaluit
│  ├── Jamaica
│  ├── Jujuy
│  ├── Juneau
│  ├── Kentucky
│  │  ├── Louisville
│  │  └── Monticello
│  ├── Knox_IN
│  ├── Kralendijk
│  ├── La_Paz
│  ├── Lima
│  ├── Los_Angeles
│  ├── Louisville
│  ├── Lower_Princes
│  ├── Maceio
│  ├── Managua
│  ├── Manaus
│  ├── Marigot
│  ├── Martinique
│  ├── Matamoros
│  ├── Mazatlan
│  ├── Mendoza
│  ├── Menominee
│  ├── Merida
│  ├── Metlakatla
│  ├── Mexico_City
│  ├── Miquelon
│  ├── Moncton
│  ├── Monterrey
│  ├── Montevideo
│  ├── Montreal
│  ├── Montserrat
│  ├── Nassau
│  ├── New_York
│  ├── Nipigon
│  ├── Nome
│  ├── Noronha
│  ├── North_Dakota
│  │  ├── Beulah
│  │  ├── Center
│  │  └── New_Salem
│  ├── Nuuk
│  ├── Ojinaga
│  ├── Panama
│  ├── Pangnirtung
│  ├── Paramaribo
│  ├── Phoenix
│  ├── Port-au-Prince
│  ├── Port_of_Spain
│  ├── Porto_Acre
│  ├── Porto_Velho
│  ├── Puerto_Rico
│  ├── Punta_Arenas
│  ├── Rainy_River
│  ├── Rankin_Inlet
│  ├── Recife
│  ├── Regina
│  ├── Resolute
│  ├── Rio_Branco
│  ├── Rosario
│  ├── Santa_Isabel
│  ├── Santarem
│  ├── Santiago
│  ├── Santo_Domingo
│  ├── Sao_Paulo
│  ├── Scoresbysund
│  ├── Shiprock
│  ├── Sitka
│  ├── St_Barthelemy
│  ├── St_Johns
│  ├── St_Kitts
│  ├── St_Lucia
│  ├── St_Thomas
│  ├── St_Vincent
│  ├── Swift_Current
│  ├── Tegucigalpa
│  ├── Thule
│  ├── Thunder_Bay
│  ├── Tijuana
│  ├── Toronto
│  ├── Tortola
│  ├── Vancouver
│  ├── Virgin
│  ├── Whitehorse
│  ├── Winnipeg
│  ├── Yakutat
│  └── Yellowknife
├── Antarctica
│  ├── Casey
│  ├── Davis
│  ├── DumontDUrville
│  ├── Macquarie
│  ├── Mawson
│  ├── McMurdo
│  ├── Palmer
│  ├── Rothera
│  ├── South_Pole
│  ├── Syowa
│  ├── Troll
│  └── Vostok
├── Arctic
│  └── Longyearbyen
├── Asia
│  ├── Aden
│  ├── Almaty
│  ├── Amman
│  ├── Anadyr
│  ├── Aqtau
│  ├── Aqtobe
│  ├── Ashgabat
│  ├── Ashkhabad
│  ├── Atyrau
│  ├── Baghdad
│  ├── Bahrain
│  ├── Baku
│  ├── Bangkok
│  ├── Barnaul
│  ├── Beirut
│  ├── Bishkek
│  ├── Brunei
│  ├── Calcutta
│  ├── Chita
│  ├── Choibalsan
│  ├── Chongqing
│  ├── Chungking
│  ├── Colombo
│  ├── Dacca
│  ├── Damascus
│  ├── Dhaka
│  ├── Dili
│  ├── Dubai
│  ├── Dushanbe
│  ├── Famagusta
│  ├── Gaza
│  ├── Harbin
│  ├── Hebron
│  ├── Ho_Chi_Minh
│  ├── Hong_Kong
│  ├── Hovd
│  ├── Irkutsk
│  ├── Istanbul
│  ├── Jakarta
│  ├── Jayapura
│  ├── Jerusalem
│  ├── Kabul
│  ├── Kamchatka
│  ├── Karachi
│  ├── Kashgar
│  ├── Kathmandu
│  ├── Katmandu
│  ├── Khandyga
│  ├── Kolkata
│  ├── Krasnoyarsk
│  ├── Kuala_Lumpur
│  ├── Kuching
│  ├── Kuwait
│  ├── Macao
│  ├── Macau
│  ├── Magadan
│  ├── Makassar
│  ├── Manila
│  ├── Muscat
│  ├── Nicosia
│  ├── Novokuznetsk
│  ├── Novosibirsk
│  ├── Omsk
│  ├── Oral
│  ├── Phnom_Penh
│  ├── Pontianak
│  ├── Pyongyang
│  ├── Qatar
│  ├── Qostanay
│  ├── Qyzylorda
│  ├── Rangoon
│  ├── Riyadh
│  ├── Saigon
│  ├── Sakhalin
│  ├── Samarkand
│  ├── Seoul
│  ├── Shanghai
│  ├── Singapore
│  ├── Srednekolymsk
│  ├── Taipei
│  ├── Tashkent
│  ├── Tbilisi
│  ├── Tehran
│  ├── Tel_Aviv
│  ├── Thimbu
│  ├── Thimphu
│  ├── Tokyo
│  ├── Tomsk
│  ├── Ujung_Pandang
│  ├── Ulaanbaatar
│  ├── Ulan_Bator
│  ├── Urumqi
│  ├── Ust-Nera
│  ├── Vientiane
│  ├── Vladivostok
│  ├── Yakutsk
│  ├── Yangon
│  ├── Yekaterinburg
│  └── Yerevan
├── Atlantic
│  ├── Azores
│  ├── Bermuda
│  ├── Canary
│  ├── Cape_Verde
│  ├── Faeroe
│  ├── Faroe
│  ├── Jan_Mayen
│  ├── Madeira
│  ├── Reykjavik
│  ├── South_Georgia
│  ├── St_Helena
│  └── Stanley
├── Australia
│  ├── ACT
│  ├── Adelaide
│  ├── Brisbane
│  ├── Broken_Hill
│  ├── Canberra
│  ├── Currie
│  ├── Darwin
│  ├── Eucla
│  ├── Hobart
│  ├── LHI
│  ├── Lindeman
│  ├── Lord_Howe
│  ├── Melbourne
│  ├── North
│  ├── NSW
│  ├── Perth
│  ├── Queensland
│  ├── South
│  ├── Sydney
│  ├── Tasmania
│  ├── Victoria
│  ├── West
│  └── Yancowinna
├── Brazil
│  ├── Acre
│  ├── DeNoronha
│  ├── East
│  └── West
├── Canada
│  ├── Atlantic
│  ├── Central
│  ├── Eastern
│  ├── Mountain
│  ├── Newfoundland
│  ├── Pacific
│  ├── Saskatchewan
│  └── Yukon
├── CET
├── Chile
│  ├── Continental
│  └── EasterIsland
├── CST6CDT
├── Cuba
├── EET
├── Egypt
├── Eire
├── EST
├── EST5EDT
├── Etc
│  ├── GMT
│  ├── GMT+0
│  ├── GMT+1
│  ├── GMT+2
│  ├── GMT+3
│  ├── GMT+4
│  ├── GMT+5
│  ├── GMT+6
│  ├── GMT+7
│  ├── GMT+8
│  ├── GMT+9
│  ├── GMT+10
│  ├── GMT+11
│  ├── GMT+12
│  ├── GMT-0
│  ├── GMT-1
│  ├── GMT-2
│  ├── GMT-3
│  ├── GMT-4
│  ├── GMT-5
│  ├── GMT-6
│  ├── GMT-7
│  ├── GMT-8
│  ├── GMT-9
│  ├── GMT-10
│  ├── GMT-11
│  ├── GMT-12
│  ├── GMT-13
│  ├── GMT-14
│  ├── GMT0
│  ├── Greenwich
│  ├── UCT
│  ├── Universal
│  ├── UTC
│  └── Zulu
├── Europe
│  ├── Amsterdam
│  ├── Andorra
│  ├── Astrakhan
│  ├── Athens
│  ├── Belfast
│  ├── Belgrade
│  ├── Berlin
│  ├── Bratislava
│  ├── Brussels
│  ├── Bucharest
│  ├── Budapest
│  ├── Busingen
│  ├── Chisinau
│  ├── Copenhagen
│  ├── Dublin
│  ├── Gibraltar
│  ├── Guernsey
│  ├── Helsinki
│  ├── Isle_of_Man
│  ├── Istanbul
│  ├── Jersey
│  ├── Kaliningrad
│  ├── Kiev
│  ├── Kirov
│  ├── Kyiv
│  ├── Lisbon
│  ├── Ljubljana
│  ├── London
│  ├── Luxembourg
│  ├── Madrid
│  ├── Malta
│  ├── Mariehamn
│  ├── Minsk
│  ├── Monaco
│  ├── Moscow
│  ├── Nicosia
│  ├── Oslo
│  ├── Paris
│  ├── Podgorica
│  ├── Prague
│  ├── Riga
│  ├── Rome
│  ├── Samara
│  ├── San_Marino
│  ├── Sarajevo
│  ├── Saratov
│  ├── Simferopol
│  ├── Skopje
│  ├── Sofia
│  ├── Stockholm
│  ├── Tallinn
│  ├── Tirane
│  ├── Tiraspol
│  ├── Ulyanovsk
│  ├── Uzhgorod
│  ├── Vaduz
│  ├── Vatican
│  ├── Vienna
│  ├── Vilnius
│  ├── Volgograd
│  ├── Warsaw
│  ├── Zagreb
│  ├── Zaporozhye
│  └── Zurich
├── Factory
├── GB
├── GB-Eire
├── GMT
├── GMT+0
├── GMT-0
├── GMT0
├── Greenwich
├── Hongkong
├── HST
├── Iceland
├── Indian
│  ├── Antananarivo
│  ├── Chagos
│  ├── Christmas
│  ├── Cocos
│  ├── Comoro
│  ├── Kerguelen
│  ├── Mahe
│  ├── Maldives
│  ├── Mauritius
│  ├── Mayotte
│  └── Reunion
├── Iran
├── iso3166.tab
├── Israel
├── Jamaica
├── Japan
├── Kwajalein
├── leap-seconds.list
├── leapseconds
├── Libya
├── MET
├── Mexico
│  ├── BajaNorte
│  ├── BajaSur
│  └── General
├── MST
├── MST7MDT
├── Navajo
├── NZ
├── NZ-CHAT
├── Pacific
│  ├── Apia
│  ├── Auckland
│  ├── Bougainville
│  ├── Chatham
│  ├── Chuuk
│  ├── Easter
│  ├── Efate
│  ├── Enderbury
│  ├── Fakaofo
│  ├── Fiji
│  ├── Funafuti
│  ├── Galapagos
│  ├── Gambier
│  ├── Guadalcanal
│  ├── Guam
│  ├── Honolulu
│  ├── Johnston
│  ├── Kanton
│  ├── Kiritimati
│  ├── Kosrae
│  ├── Kwajalein
│  ├── Majuro
│  ├── Marquesas
│  ├── Midway
│  ├── Nauru
│  ├── Niue
│  ├── Norfolk
│  ├── Noumea
│  ├── Pago_Pago
│  ├── Palau
│  ├── Pitcairn
│  ├── Pohnpei
│  ├── Ponape
│  ├── Port_Moresby
│  ├── Rarotonga
│  ├── Saipan
│  ├── Samoa
│  ├── Tahiti
│  ├── Tarawa
│  ├── Tongatapu
│  ├── Truk
│  ├── Wake
│  ├── Wallis
│  └── Yap
├── Poland
├── Portugal
├── posix
│  ├── Africa -> ../Africa
│  ├── America -> ../America
│  ├── Antarctica -> ../Antarctica
│  ├── Arctic -> ../Arctic
│  ├── Asia -> ../Asia
│  ├── Atlantic -> ../Atlantic
│  ├── Australia -> ../Australia
│  ├── Brazil -> ../Brazil
│  ├── Canada -> ../Canada
│  ├── CET -> ../CET
│  ├── Chile -> ../Chile
│  ├── CST6CDT -> ../CST6CDT
│  ├── Cuba -> ../Cuba
│  ├── EET -> ../EET
│  ├── Egypt -> ../Egypt
│  ├── Eire -> ../Eire
│  ├── EST -> ../EST
│  ├── EST5EDT -> ../EST5EDT
│  ├── Etc -> ../Etc
│  ├── Europe -> ../Europe
│  ├── Factory -> ../Factory
│  ├── GB -> ../GB
│  ├── GB-Eire -> ../GB-Eire
│  ├── GMT -> ../GMT
│  ├── GMT+0 -> ../GMT+0
│  ├── GMT-0 -> ../GMT-0
│  ├── GMT0 -> ../GMT0
│  ├── Greenwich -> ../Greenwich
│  ├── Hongkong -> ../Hongkong
│  ├── HST -> ../HST
│  ├── Iceland -> ../Iceland
│  ├── Indian -> ../Indian
│  ├── Iran -> ../Iran
│  ├── iso3166.tab -> ../iso3166.tab
│  ├── Israel -> ../Israel
│  ├── Jamaica -> ../Jamaica
│  ├── Japan -> ../Japan
│  ├── Kwajalein -> ../Kwajalein
│  ├── leapseconds -> ../leapseconds
│  ├── Libya -> ../Libya
│  ├── MET -> ../MET
│  ├── Mexico -> ../Mexico
│  ├── MST -> ../MST
│  ├── MST7MDT -> ../MST7MDT
│  ├── Navajo -> ../Navajo
│  ├── NZ -> ../NZ
│  ├── NZ-CHAT -> ../NZ-CHAT
│  ├── Pacific -> ../Pacific
│  ├── Poland -> ../Poland
│  ├── Portugal -> ../Portugal
│  ├── PRC -> ../PRC
│  ├── PST8PDT -> ../PST8PDT
│  ├── ROC -> ../ROC
│  ├── ROK -> ../ROK
│  ├── Singapore -> ../Singapore
│  ├── Turkey -> ../Turkey
│  ├── tzdata.zi -> ../tzdata.zi
│  ├── UCT -> ../UCT
│  ├── Universal -> ../Universal
│  ├── US -> ../US
│  ├── UTC -> ../UTC
│  ├── W-SU -> ../W-SU
│  ├── WET -> ../WET
│  ├── zone.tab -> ../zone.tab
│  ├── zone1970.tab -> ../zone1970.tab
│  ├── zonenow.tab -> ../zonenow.tab
│  └── Zulu -> ../Zulu
├── PRC
├── PST8PDT
├── right
│  ├── Africa
│  │  ├── Abidjan
│  │  ├── Accra
│  │  ├── Addis_Ababa
│  │  ├── Algiers
│  │  ├── Asmara
│  │  ├── Asmera
│  │  ├── Bamako
│  │  ├── Bangui
│  │  ├── Banjul
│  │  ├── Bissau
│  │  ├── Blantyre
│  │  ├── Brazzaville
│  │  ├── Bujumbura
│  │  ├── Cairo
│  │  ├── Casablanca
│  │  ├── Ceuta
│  │  ├── Conakry
│  │  ├── Dakar
│  │  ├── Dar_es_Salaam
│  │  ├── Djibouti
│  │  ├── Douala
│  │  ├── El_Aaiun
│  │  ├── Freetown
│  │  ├── Gaborone
│  │  ├── Harare
│  │  ├── Johannesburg
│  │  ├── Juba
│  │  ├── Kampala
│  │  ├── Khartoum
│  │  ├── Kigali
│  │  ├── Kinshasa
│  │  ├── Lagos
│  │  ├── Libreville
│  │  ├── Lome
│  │  ├── Luanda
│  │  ├── Lubumbashi
│  │  ├── Lusaka
│  │  ├── Malabo
│  │  ├── Maputo
│  │  ├── Maseru
│  │  ├── Mbabane
│  │  ├── Mogadishu
│  │  ├── Monrovia
│  │  ├── Nairobi
│  │  ├── Ndjamena
│  │  ├── Niamey
│  │  ├── Nouakchott
│  │  ├── Ouagadougou
│  │  ├── Porto-Novo
│  │  ├── Sao_Tome
│  │  ├── Timbuktu
│  │  ├── Tripoli
│  │  ├── Tunis
│  │  └── Windhoek
│  ├── America
│  │  ├── Adak
│  │  ├── Anchorage
│  │  ├── Anguilla
│  │  ├── Antigua
│  │  ├── Araguaina
│  │  ├── Argentina
│  │  │  ├── Buenos_Aires
│  │  │  ├── Catamarca
│  │  │  ├── ComodRivadavia
│  │  │  ├── Cordoba
│  │  │  ├── Jujuy
│  │  │  ├── La_Rioja
│  │  │  ├── Mendoza
│  │  │  ├── Rio_Gallegos
│  │  │  ├── Salta
│  │  │  ├── San_Juan
│  │  │  ├── San_Luis
│  │  │  ├── Tucuman
│  │  │  └── Ushuaia
│  │  ├── Aruba
│  │  ├── Asuncion
│  │  ├── Atikokan
│  │  ├── Atka
│  │  ├── Bahia
│  │  ├── Bahia_Banderas
│  │  ├── Barbados
│  │  ├── Belem
│  │  ├── Belize
│  │  ├── Blanc-Sablon
│  │  ├── Boa_Vista
│  │  ├── Bogota
│  │  ├── Boise
│  │  ├── Buenos_Aires
│  │  ├── Cambridge_Bay
│  │  ├── Campo_Grande
│  │  ├── Cancun
│  │  ├── Caracas
│  │  ├── Catamarca
│  │  ├── Cayenne
│  │  ├── Cayman
│  │  ├── Chicago
│  │  ├── Chihuahua
│  │  ├── Ciudad_Juarez
│  │  ├── Coral_Harbour
│  │  ├── Cordoba
│  │  ├── Costa_Rica
│  │  ├── Creston
│  │  ├── Cuiaba
│  │  ├── Curacao
│  │  ├── Danmarkshavn
│  │  ├── Dawson
│  │  ├── Dawson_Creek
│  │  ├── Denver
│  │  ├── Detroit
│  │  ├── Dominica
│  │  ├── Edmonton
│  │  ├── Eirunepe
│  │  ├── El_Salvador
│  │  ├── Ensenada
│  │  ├── Fort_Nelson
│  │  ├── Fort_Wayne
│  │  ├── Fortaleza
│  │  ├── Glace_Bay
│  │  ├── Godthab
│  │  ├── Goose_Bay
│  │  ├── Grand_Turk
│  │  ├── Grenada
│  │  ├── Guadeloupe
│  │  ├── Guatemala
│  │  ├── Guayaquil
│  │  ├── Guyana
│  │  ├── Halifax
│  │  ├── Havana
│  │  ├── Hermosillo
│  │  ├── Indiana
│  │  │  ├── Indianapolis
│  │  │  ├── Knox
│  │  │  ├── Marengo
│  │  │  ├── Petersburg
│  │  │  ├── Tell_City
│  │  │  ├── Vevay
│  │  │  ├── Vincennes
│  │  │  └── Winamac
│  │  ├── Indianapolis
│  │  ├── Inuvik
│  │  ├── Iqaluit
│  │  ├── Jamaica
│  │  ├── Jujuy
│  │  ├── Juneau
│  │  ├── Kentucky
│  │  │  ├── Louisville
│  │  │  └── Monticello
│  │  ├── Knox_IN
│  │  ├── Kralendijk
│  │  ├── La_Paz
│  │  ├── Lima
│  │  ├── Los_Angeles
│  │  ├── Louisville
│  │  ├── Lower_Princes
│  │  ├── Maceio
│  │  ├── Managua
│  │  ├── Manaus
│  │  ├── Marigot
│  │  ├── Martinique
│  │  ├── Matamoros
│  │  ├── Mazatlan
│  │  ├── Mendoza
│  │  ├── Menominee
│  │  ├── Merida
│  │  ├── Metlakatla
│  │  ├── Mexico_City
│  │  ├── Miquelon
│  │  ├── Moncton
│  │  ├── Monterrey
│  │  ├── Montevideo
│  │  ├── Montreal
│  │  ├── Montserrat
│  │  ├── Nassau
│  │  ├── New_York
│  │  ├── Nipigon
│  │  ├── Nome
│  │  ├── Noronha
│  │  ├── North_Dakota
│  │  │  ├── Beulah
│  │  │  ├── Center
│  │  │  └── New_Salem
│  │  ├── Nuuk
│  │  ├── Ojinaga
│  │  ├── Panama
│  │  ├── Pangnirtung
│  │  ├── Paramaribo
│  │  ├── Phoenix
│  │  ├── Port-au-Prince
│  │  ├── Port_of_Spain
│  │  ├── Porto_Acre
│  │  ├── Porto_Velho
│  │  ├── Puerto_Rico
│  │  ├── Punta_Arenas
│  │  ├── Rainy_River
│  │  ├── Rankin_Inlet
│  │  ├── Recife
│  │  ├── Regina
│  │  ├── Resolute
│  │  ├── Rio_Branco
│  │  ├── Rosario
│  │  ├── Santa_Isabel
│  │  ├── Santarem
│  │  ├── Santiago
│  │  ├── Santo_Domingo
│  │  ├── Sao_Paulo
│  │  ├── Scoresbysund
│  │  ├── Shiprock
│  │  ├── Sitka
│  │  ├── St_Barthelemy
│  │  ├── St_Johns
│  │  ├── St_Kitts
│  │  ├── St_Lucia
│  │  ├── St_Thomas
│  │  ├── St_Vincent
│  │  ├── Swift_Current
│  │  ├── Tegucigalpa
│  │  ├── Thule
│  │  ├── Thunder_Bay
│  │  ├── Tijuana
│  │  ├── Toronto
│  │  ├── Tortola
│  │  ├── Vancouver
│  │  ├── Virgin
│  │  ├── Whitehorse
│  │  ├── Winnipeg
│  │  ├── Yakutat
│  │  └── Yellowknife
│  ├── Antarctica
│  │  ├── Casey
│  │  ├── Davis
│  │  ├── DumontDUrville
│  │  ├── Macquarie
│  │  ├── Mawson
│  │  ├── McMurdo
│  │  ├── Palmer
│  │  ├── Rothera
│  │  ├── South_Pole
│  │  ├── Syowa
│  │  ├── Troll
│  │  └── Vostok
│  ├── Arctic
│  │  └── Longyearbyen
│  ├── Asia
│  │  ├── Aden
│  │  ├── Almaty
│  │  ├── Amman
│  │  ├── Anadyr
│  │  ├── Aqtau
│  │  ├── Aqtobe
│  │  ├── Ashgabat
│  │  ├── Ashkhabad
│  │  ├── Atyrau
│  │  ├── Baghdad
│  │  ├── Bahrain
│  │  ├── Baku
│  │  ├── Bangkok
│  │  ├── Barnaul
│  │  ├── Beirut
│  │  ├── Bishkek
│  │  ├── Brunei
│  │  ├── Calcutta
│  │  ├── Chita
│  │  ├── Choibalsan
│  │  ├── Chongqing
│  │  ├── Chungking
│  │  ├── Colombo
│  │  ├── Dacca
│  │  ├── Damascus
│  │  ├── Dhaka
│  │  ├── Dili
│  │  ├── Dubai
│  │  ├── Dushanbe
│  │  ├── Famagusta
│  │  ├── Gaza
│  │  ├── Harbin
│  │  ├── Hebron
│  │  ├── Ho_Chi_Minh
│  │  ├── Hong_Kong
│  │  ├── Hovd
│  │  ├── Irkutsk
│  │  ├── Istanbul
│  │  ├── Jakarta
│  │  ├── Jayapura
│  │  ├── Jerusalem
│  │  ├── Kabul
│  │  ├── Kamchatka
│  │  ├── Karachi
│  │  ├── Kashgar
│  │  ├── Kathmandu
│  │  ├── Katmandu
│  │  ├── Khandyga
│  │  ├── Kolkata
│  │  ├── Krasnoyarsk
│  │  ├── Kuala_Lumpur
│  │  ├── Kuching
│  │  ├── Kuwait
│  │  ├── Macao
│  │  ├── Macau
│  │  ├── Magadan
│  │  ├── Makassar
│  │  ├── Manila
│  │  ├── Muscat
│  │  ├── Nicosia
│  │  ├── Novokuznetsk
│  │  ├── Novosibirsk
│  │  ├── Omsk
│  │  ├── Oral
│  │  ├── Phnom_Penh
│  │  ├── Pontianak
│  │  ├── Pyongyang
│  │  ├── Qatar
│  │  ├── Qostanay
│  │  ├── Qyzylorda
│  │  ├── Rangoon
│  │  ├── Riyadh
│  │  ├── Saigon
│  │  ├── Sakhalin
│  │  ├── Samarkand
│  │  ├── Seoul
│  │  ├── Shanghai
│  │  ├── Singapore
│  │  ├── Srednekolymsk
│  │  ├── Taipei
│  │  ├── Tashkent
│  │  ├── Tbilisi
│  │  ├── Tehran
│  │  ├── Tel_Aviv
│  │  ├── Thimbu
│  │  ├── Thimphu
│  │  ├── Tokyo
│  │  ├── Tomsk
│  │  ├── Ujung_Pandang
│  │  ├── Ulaanbaatar
│  │  ├── Ulan_Bator
│  │  ├── Urumqi
│  │  ├── Ust-Nera
│  │  ├── Vientiane
│  │  ├── Vladivostok
│  │  ├── Yakutsk
│  │  ├── Yangon
│  │  ├── Yekaterinburg
│  │  └── Yerevan
│  ├── Atlantic
│  │  ├── Azores
│  │  ├── Bermuda
│  │  ├── Canary
│  │  ├── Cape_Verde
│  │  ├── Faeroe
│  │  ├── Faroe
│  │  ├── Jan_Mayen
│  │  ├── Madeira
│  │  ├── Reykjavik
│  │  ├── South_Georgia
│  │  ├── St_Helena
│  │  └── Stanley
│  ├── Australia
│  │  ├── ACT
│  │  ├── Adelaide
│  │  ├── Brisbane
│  │  ├── Broken_Hill
│  │  ├── Canberra
│  │  ├── Currie
│  │  ├── Darwin
│  │  ├── Eucla
│  │  ├── Hobart
│  │  ├── LHI
│  │  ├── Lindeman
│  │  ├── Lord_Howe
│  │  ├── Melbourne
│  │  ├── North
│  │  ├── NSW
│  │  ├── Perth
│  │  ├── Queensland
│  │  ├── South
│  │  ├── Sydney
│  │  ├── Tasmania
│  │  ├── Victoria
│  │  ├── West
│  │  └── Yancowinna
│  ├── Brazil
│  │  ├── Acre
│  │  ├── DeNoronha
│  │  ├── East
│  │  └── West
│  ├── Canada
│  │  ├── Atlantic
│  │  ├── Central
│  │  ├── Eastern
│  │  ├── Mountain
│  │  ├── Newfoundland
│  │  ├── Pacific
│  │  ├── Saskatchewan
│  │  └── Yukon
│  ├── CET
│  ├── Chile
│  │  ├── Continental
│  │  └── EasterIsland
│  ├── CST6CDT
│  ├── Cuba
│  ├── EET
│  ├── Egypt
│  ├── Eire
│  ├── EST
│  ├── EST5EDT
│  ├── Etc
│  │  ├── GMT
│  │  ├── GMT+0
│  │  ├── GMT+1
│  │  ├── GMT+2
│  │  ├── GMT+3
│  │  ├── GMT+4
│  │  ├── GMT+5
│  │  ├── GMT+6
│  │  ├── GMT+7
│  │  ├── GMT+8
│  │  ├── GMT+9
│  │  ├── GMT+10
│  │  ├── GMT+11
│  │  ├── GMT+12
│  │  ├── GMT-0
│  │  ├── GMT-1
│  │  ├── GMT-2
│  │  ├── GMT-3
│  │  ├── GMT-4
│  │  ├── GMT-5
│  │  ├── GMT-6
│  │  ├── GMT-7
│  │  ├── GMT-8
│  │  ├── GMT-9
│  │  ├── GMT-10
│  │  ├── GMT-11
│  │  ├── GMT-12
│  │  ├── GMT-13
│  │  ├── GMT-14
│  │  ├── GMT0
│  │  ├── Greenwich
│  │  ├── UCT
│  │  ├── Universal
│  │  ├── UTC
│  │  └── Zulu
│  ├── Europe
│  │  ├── Amsterdam
│  │  ├── Andorra
│  │  ├── Astrakhan
│  │  ├── Athens
│  │  ├── Belfast
│  │  ├── Belgrade
│  │  ├── Berlin
│  │  ├── Bratislava
│  │  ├── Brussels
│  │  ├── Bucharest
│  │  ├── Budapest
│  │  ├── Busingen
│  │  ├── Chisinau
│  │  ├── Copenhagen
│  │  ├── Dublin
│  │  ├── Gibraltar
│  │  ├── Guernsey
│  │  ├── Helsinki
│  │  ├── Isle_of_Man
│  │  ├── Istanbul
│  │  ├── Jersey
│  │  ├── Kaliningrad
│  │  ├── Kiev
│  │  ├── Kirov
│  │  ├── Kyiv
│  │  ├── Lisbon
│  │  ├── Ljubljana
│  │  ├── London
│  │  ├── Luxembourg
│  │  ├── Madrid
│  │  ├── Malta
│  │  ├── Mariehamn
│  │  ├── Minsk
│  │  ├── Monaco
│  │  ├── Moscow
│  │  ├── Nicosia
│  │  ├── Oslo
│  │  ├── Paris
│  │  ├── Podgorica
│  │  ├── Prague
│  │  ├── Riga
│  │  ├── Rome
│  │  ├── Samara
│  │  ├── San_Marino
│  │  ├── Sarajevo
│  │  ├── Saratov
│  │  ├── Simferopol
│  │  ├── Skopje
│  │  ├── Sofia
│  │  ├── Stockholm
│  │  ├── Tallinn
│  │  ├── Tirane
│  │  ├── Tiraspol
│  │  ├── Ulyanovsk
│  │  ├── Uzhgorod
│  │  ├── Vaduz
│  │  ├── Vatican
│  │  ├── Vienna
│  │  ├── Vilnius
│  │  ├── Volgograd
│  │  ├── Warsaw
│  │  ├── Zagreb
│  │  ├── Zaporozhye
│  │  └── Zurich
│  ├── Factory
│  ├── GB
│  ├── GB-Eire
│  ├── GMT
│  ├── GMT+0
│  ├── GMT-0
│  ├── GMT0
│  ├── Greenwich
│  ├── Hongkong
│  ├── HST
│  ├── Iceland
│  ├── Indian
│  │  ├── Antananarivo
│  │  ├── Chagos
│  │  ├── Christmas
│  │  ├── Cocos
│  │  ├── Comoro
│  │  ├── Kerguelen
│  │  ├── Mahe
│  │  ├── Maldives
│  │  ├── Mauritius
│  │  ├── Mayotte
│  │  └── Reunion
│  ├── Iran
│  ├── Israel
│  ├── Jamaica
│  ├── Japan
│  ├── Kwajalein
│  ├── Libya
│  ├── MET
│  ├── Mexico
│  │  ├── BajaNorte
│  │  ├── BajaSur
│  │  └── General
│  ├── MST
│  ├── MST7MDT
│  ├── Navajo
│  ├── NZ
│  ├── NZ-CHAT
│  ├── Pacific
│  │  ├── Apia
│  │  ├── Auckland
│  │  ├── Bougainville
│  │  ├── Chatham
│  │  ├── Chuuk
│  │  ├── Easter
│  │  ├── Efate
│  │  ├── Enderbury
│  │  ├── Fakaofo
│  │  ├── Fiji
│  │  ├── Funafuti
│  │  ├── Galapagos
│  │  ├── Gambier
│  │  ├── Guadalcanal
│  │  ├── Guam
│  │  ├── Honolulu
│  │  ├── Johnston
│  │  ├── Kanton
│  │  ├── Kiritimati
│  │  ├── Kosrae
│  │  ├── Kwajalein
│  │  ├── Majuro
│  │  ├── Marquesas
│  │  ├── Midway
│  │  ├── Nauru
│  │  ├── Niue
│  │  ├── Norfolk
│  │  ├── Noumea
│  │  ├── Pago_Pago
│  │  ├── Palau
│  │  ├── Pitcairn
│  │  ├── Pohnpei
│  │  ├── Ponape
│  │  ├── Port_Moresby
│  │  ├── Rarotonga
│  │  ├── Saipan
│  │  ├── Samoa
│  │  ├── Tahiti
│  │  ├── Tarawa
│  │  ├── Tongatapu
│  │  ├── Truk
│  │  ├── Wake
│  │  ├── Wallis
│  │  └── Yap
│  ├── Poland
│  ├── Portugal
│  ├── PRC
│  ├── PST8PDT
│  ├── ROC
│  ├── ROK
│  ├── Singapore
│  ├── Turkey
│  ├── UCT
│  ├── Universal
│  ├── US
│  │  ├── Alaska
│  │  ├── Aleutian
│  │  ├── Arizona
│  │  ├── Central
│  │  ├── East-Indiana
│  │  ├── Eastern
│  │  ├── Hawaii
│  │  ├── Indiana-Starke
│  │  ├── Michigan
│  │  ├── Mountain
│  │  ├── Pacific
│  │  └── Samoa
│  ├── UTC
│  ├── W-SU
│  ├── WET
│  └── Zulu
├── ROC
├── ROK
├── Singapore
├── Turkey
├── tzdata.zi
├── UCT
├── Universal
├── US
│  ├── Alaska
│  ├── Aleutian
│  ├── Arizona
│  ├── Central
│  ├── East-Indiana
│  ├── Eastern
│  ├── Hawaii
│  ├── Indiana-Starke
│  ├── Michigan
│  ├── Mountain
│  ├── Pacific
│  └── Samoa
├── UTC
├── W-SU
├── WET
├── zone.tab
├── zone1970.tab
├── zonenow.tab
└── Zulu
```

</details>